pulp_2to3_migration/app/pre_migration.py

```diff
@@ -617,50 +617,49 @@ def handle_outdated_resources(plan, type_to_repo_ids):
         type_to_repo_ids(dict): A mapping from a pulp 2 repo type to a list of pulp 2 repo_ids
     """
     for plugin_plan in plan.get_plugin_plans():
-        repos = plugin_plan.get_repositories()
+        inplan_repos = plugin_plan.get_repositories()
 
         # filter by repo type
         repos_to_consider = type_to_repo_ids[plugin_plan.type]
 
         # in case only certain repositories are specified in the migration plan
-        if repos:
-            repos_to_consider = set(repos).intersection(repos_to_consider)
+        if inplan_repos:
+            repos_to_consider = set(inplan_repos).intersection(repos_to_consider)
 
         mongo_repo_q = mongo_Q(repo_id__in=repos_to_consider)
+        mongo_repo_obj_ids = set(str(i.id) for i in Repository.objects(mongo_repo_q).only('id'))
 
-        mongo_repo_object_ids = set(
-            str(i.id) for i in Repository.objects(mongo_repo_q).only('id'))
+        repo_type_q = Q(pulp2_repo_type=plugin_plan.type)
+        inplan_repo_q = Q(pulp2_object_id__in=mongo_repo_obj_ids)
+        Pulp2Repository.objects.filter(repo_type_q).exclude(inplan_repo_q).update(not_in_plan=True)
```

Comment:
Logic for the repos  didn't change, I just simplified queries and made it in the same way as importers and distributors.